unittest/CMakeLists.txt

```diff
@@ -129,7 +129,16 @@ function(ADD_PINOCCHIO_UNIT_TEST name)
     INCLUDE_DIRS ${Boost_INCLUDE_DIRS})
 
   if(PKGS)
-    target_link_libraries(${TEST_NAME} PRIVATE ${PKGS})
+    # Convert package names to proper CMake targets
+    set(CONVERTED_PKGS "")
+    foreach(PKG IN LISTS PKGS)
+      if(PKG STREQUAL "casadi")
+        list(APPEND CONVERTED_PKGS "casadi::casadi")
+      else()
+        list(APPEND CONVERTED_PKGS ${PKG})
+      endif()
+    endforeach()
+    target_link_libraries(${TEST_NAME} PRIVATE ${CONVERTED_PKGS})
```

Comment:
Instead of doing that, look for all `PACKAGE` argument and replace casadi by ${CASADI_TARGET} ([Like in this one](https://github.com/stack-of-tasks/pinocchio/blob/devel/unittest/casadi/CMakeLists.txt#L6))

This avoid adding more complexity in CMake.